### PR TITLE
Issue 971 duplication

### PR DESCRIPTION
**Pull request by DanielSauve**
# Description 

Cleaned up the code duplication identified by code climate, mainly wrote functions to generate objects that are used in the UI since many of the UI fields are similar and popped up as duplicated code.

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/971

# Test Coverage

✅

### CODE REVIEW

File: app/helpers/medications_helper.rb [hunk 1]
```diff
@@ -229,6 +143,50 @@ def days_checkbox
...
-end
 
+  # rubocop:disable MethodLength, LineLength
+  def medication_unit_field(type)
+    {
+      id: "medication_#{type}_unit",
+      type: 'select',
+      name: "medication[#{type}_unit]",
+      dark: true,
+      value: @medication["#{type}_unit"] || t('medications.units.tablets.other'),
```
Comment (julianguyen) [inline app/helpers/medications_helper.rb]: I would just put `t('medications.units.tablets.other'),` on the next line and remove the disable LineLength comment.

File: app/helpers/viewers_helper.rb [hunk 2]
```diff
@@ -49,4 +49,32 @@ def get_viewers(data, data_type, obj)
     end
     []
   end
+
+  # rubocop:disable MethodLength
+  def get_viewers_input(viewers, name, translation_name, obj)
+    input = {}
+    if viewers.present?
+      checkboxes = []
```
Comment (julianguyen) [inline app/helpers/viewers_helper.rb]: We should move the logic for populating checkboxes into another helper.

File: config/locales/de.yml [hunk 3]
```diff
@@ -211,6 +211,7 @@ de:
    refill_hint: 'Wann dieses Rezept erneuert werden muss'
    refill_reminder_hint: "Wir senden dir eine Erinnerungsmail in der Woche bevor dieses Arzneimittel aufgefüllt werden muss."
    daily_reminder_hint: "Wir senden dir eine tägliche Erinnerungsmail zum Einnehmen dieses Arzneimittels"
+   total: 'Gesamt'
```
[the remark this answers is not in the document]
Comment (DanielSauve) [reply config/locales/de.yml]: The main reason I added the new translation was due to the former translation having a ":" at the end, while every other title field on the form doesn't contain a ":"

[the remark this answers is not in the document]
Comment (julianguyen) [reply config/locales/de.yml]: Ah, I see! In a recent commit that was merged to master, we actually remove the colon from the total. Can you try rebasing and using that existing value instead?

[the remark this answers is not in the document]
Comment (julianguyen) [reply config/locales/de.yml]: Oh it looks like `total: 'Gesamt:'` still exists
You can just remove the colon from here!

File: app/helpers/moments_helper.rb [hunk 4]
```diff
@@ -90,7 +90,7 @@ def moment_form_inputs
         checkboxes: checkboxes_for(@strategies),
         formProps: quick_create_strategy_props
       },
-      moments_viewers_input,
+      get_viewers_input(@viewers, 'moment', 'moments', @moment),
```
Comment (jm96441n) [inline app/helpers/moments_helper.rb]: Are these ivars coming from the controller? If so can we pass the ivars in from the view file as arguments to the method that calls this? [This blog post](http://www.carlosramireziii.com/stop-using-instance-variables-in-partials.html) references using ivars in partials, but I feel the same logic applies to helpers as well

**Comment by julianguyen** [comment]
Sorry for the merge conflicts with the translation files, I recently reformatted them. Let me know if you run into any problems, happy to help!

**Comment by DanielSauve** [comment]
I'm going to look at cleaning up all the things addressed by the feedback, should be done by the end of tonight!

**Comment by DanielSauve** [comment]
Sorry about this taking a little bit longer than expected, I haven't had access to my Mac the past couple days, and you were right about the Windows setup being difficult 😛

**Comment by DanielSauve** [comment]
I'll address the last two comments when I get a chance to use my Mac again 😅

File: app/helpers/viewers_helper.rb (hunk 2 above)
Comment (DanielSauve) [reply app/helpers/viewers_helper.rb]: How would you recommend I go about doing this?

Comment (julianguyen) [reply app/helpers/viewers_helper.rb]: I would make a helper method that returns the value of checkboxes and it would contain lines 58-65.

**Comment by DanielSauve** [comment]
I'm not sure why merging the changes from the upstream master into my PR caused some of the tests to fail, any ideas on what went wrong?

**Comment by julianguyen** [comment]
@DanielSauve This might be an issue that's only happening in the Circle build. I'm also encountering this in https://github.com/ifmeorg/ifme/pull/1177/files. Looking into it now. When you run the tests locally do they pass for you?

**Comment by julianguyen** [comment]
@DanielSauve I fixed the issue, can you rebase from master? :)

**Comment by DanielSauve** [comment]
Perfect! That seems to have fixed it!

**Pull request merged**